infrastructure/smart-contracts/contracts/upgrades/node-license/NodeLicense8.sol

```diff
@@ -837,6 +824,53 @@ contract NodeLicense8 is ERC721EnumerableUpgradeable, AccessControlUpgradeable
         uint256[] memory tokenIds,
         bytes32 transferId
     ) external onlyRole(TRANSFER_ROLE) {
-        revert("Not implemented");
+        require(to != address(0) && to != address(this), "Invalid to address");
+        require(!usedTransferIds[transferId], "TransferId in use");
+
+        uint256 tokenIdsLength = tokenIds.length;
+        require(tokenIdsLength > 0, "No tokenIds provided");
+
+        usedTransferIds[transferId] = true;
+        
+        for(uint256 i; i < tokenIds.length; i++){
+            super.safeTransferFrom(msg.sender, to, tokenIds[i]);
+        }
+        emit AdminTransferBatch(msg.sender, to, transferId, tokenIds);
+    }
+
+    /**
+     * @notice Overwrite ERC721 tranferFrom require TRANSFER_ROLE on msg.sender
+     */
+    function transferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public override onlyRole(TRANSFER_ROLE) {
```

Comment:
I get an IDE error here `Function needs to specify overridden contracts "ERC721Upgradeable" and "IERC721Upgradeable".`

Reply:
I would guess this has something to do with the IDE rather than the solidity. I managed to compile without error and the test suite seems to work fine with it. We will know for sure once deploying to sepolia.